### PR TITLE
Refactor setupTimer to retimer

### PR DESCRIPTION
**Pull request by behrad**
keepalive re-scheduling is done via retimer instead of setTimeout.

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -164,10 +162,14 @@ Client.prototype.setUpTimer = function() {
 
   this.logger.debug({ timeout: timeout }, "setting keepalive timeout");
 
-  this.timer = setTimeout(function() {
-    that.logger.info("keepalive timeout");
-    that.onNonDisconnectClose();
-  }, timeout);
+  if( this.timer ) {
```
Comment (mcollina) [inline lib/client.js]: can you write this one as `if (this.timer)`?

File: lib/client.js [hunk 2]
```diff
@@ -164,10 +162,14 @@ Client.prototype.setUpTimer = function() {
...
+    this.timer.reschedule( timeout );
```
Comment (mcollina) [inline lib/client.js]: and this one as `this.timer.reschedule(timeout);`

**Comment by mcollina** [comment]
Apart from minor linting, this is good to go :).

**Comment by mcollina** [comment]
Have you had the chance to see how much this impact performance?

**Comment by behrad** [comment]
testing...

**Pull request merged**